providers/dns/joker/client.go

```diff
@@ -70,7 +70,20 @@ func (d *DNSProvider) login() (*response, error) {
 		return nil, nil
 	}
 
-	response, err := d.postRequest("login", url.Values{"api-key": {d.config.APIKey}})
+	var values url.Values
+	switch {
+	case d.config.Username != "" && d.config.Password != "":
```

Comment:
Personally not a fan of using switch like this rather than `if {} else if {} else {}`, but this isn't a blocker.